packer-template/hcloud-leapmicro-snapshots.pkr.hcl

```
@@ -0,0 +1,245 @@
+}
+
+# We download the OpenSUSE Leap Micro ARM image from an automatically selected mirror.
+variable "opensuse_leapmicro_arm_mirror_link" {
+  type    = string
+  default = "https://download.opensuse.org/distribution/leap-micro/6.1/appliances/openSUSE-Leap-Micro.aarch64-Base-qcow.qcow2"
+}
+
+# If you need to add other packages to the OS, do it here in the default value, like ["vim", "curl", "wget"]
+# When looking for packages, you need to search for OpenSUSE Tumbleweed packages, as Leap Micro is based on Tumbleweed.
+variable "packages_to_install" {
+  type    = list(string)
+  default = []
+}
+
+locals {
+  needed_packages = join(" ", concat([
+    "busybox-bzip2",
+    "container-selinux",
+    "policycoreutils",
+    "policycoreutils-devel",
+    "policycoreutils-python-utils",
+    "python3-policycoreutils",
+    "python311-setools",
+    "restorecond",
+    "selinux-policy",
+    "fuse-overlayfs",
+    #"fuse",
+    #"libfuse3",
+    "audit",
+    "open-iscsi",
+    "nfs-client",
+    "xfsprogs",
+    "cryptsetup",
+    "lvm2",
+    "git",
+    "cifs-utils",
+    "bash-completion",
+    "udica"
+  ], var.packages_to_install))
+
+  # leap_oss_packages = join(" ", [
+  #   #"bind-utils",
+  #   "bzip2",
+  #   "wireguard-tools",
+  #   "mtr",
+  #   "tcpdump"
+  # ])
+
+  # Add local variables for inline shell commands
+  download_image = "wget --timeout=5 --waitretry=5 --tries=5 --retry-connrefused --inet4-only "
+
+  write_image = <<-EOT
+    set -ex
+    echo 'Leap Micro image loaded, writing to disk... '
+    qemu-img convert -p -f qcow2 -O host_device $(ls -a | grep -ie '^opensuse.*leap-micro.*qcow2$') /dev/sda
+    echo 'done. Rebooting...'
+    sleep 1 && udevadm settle && reboot
+  EOT
+
+  install_packages = <<-EOT
+    set -ex
+
+    # echo "First reboot successful, installing needed packages..."
+    ARCH=$(uname -m)
+    REPO_URL="https://ftp.rz.uni-wuerzburg.de/opensuse/distribution/leap-micro/6.1/product/repo/openSUSE-Leap-Micro-6.1-$ARCH"
+    transactional-update shell <<- EOF
+    setenforce 0
+
+    # Added mirror with higher prio.
+    # As of now (23.04.2025), the main repo doesn't have container-selinux in vers. 2.236
+    # thereforce, we prioritize the mirrors, as they seem to have the proper versions
+
+    zypper ar "$REPO_URL" uni-wuerzburg-mirror
+    zypper --non-interactive --gpg-auto-import-keys refresh
+
+    zypper --verbose --non-interactive install --allow-vendor-change ${local.needed_packages}
+
+    rpm --import https://rpm.rancher.io/public.key
+    zypper --verbose --non-interactive install -y https://github.com/k3s-io/k3s-selinux/releases/download/v1.6.stable.1/k3s-selinux-1.6-1.slemicro.noarch.rpm
+    zypper addlock k3s-selinux
+
+    restorecon -Rv /etc/selinux/targeted/policy
+    restorecon -Rv /var/lib
+    fixfiles restore
+    touch /.autorelabel
+
+    echo so what do we have here
+    rpm -qa |grep container
+    zypper search-packages -s container-selinux
+
```

```suggestion
    zypper search-packages -s container-selinux
    zypper update -y

```

Otherwise the snapshot is already super outdated during creation. FWIW, I tested the update interactively and it will download packages in size of 1GB (as we already 5 months after the initial release).